src/content/en/updates/2020/12/webassembly.md

```diff
@@ -0,0 +1,430 @@
+everything looks good (for example, that “alpha” channel is always set
+to full opacity). Similarly, we can expand and check the real and
+imaginary parts of the complex number stored in the `center` variable.
+
+If you want to access a deeply nested property that is otherwise hard to
+navigate to via the **Scope** view, you can use the **Console**
+evaluation, too! However, note that more complex C++ expressions are not
+yet supported.
+
+![Console panel showing the result of `palette[10].r`](/web/updates/images/2020/12/webassembly/image12.png)
+
+Let’s resume execution a few times and we can see how the inner `x` is
+changing as well – either by looking in the **Scope** view again, adding
+the variable name to the watch list, evaluating it in the console, or by
+hovering over the variable in the source code:
+
+![Tooltip over the variable `x` in the source showing its value `3`](/web/updates/images/2020/12/webassembly/image10.png)
+
+From here, we can step-in or step-over C++ statements, and observe how
+other variables are changing too:
+
+![Tooltips and Scope view showing values of `color`, `point` and other variables](/web/updates/images/2020/12/webassembly/image8.png)
+
+Okay, so this all works great when a debug information is available, but
+what if we want to debug a code that wasn’t built with the debugging
+options?
+
+## Raw WebAssembly debugging {: #raw }
+
+For example, we asked Emscripten to provide a prebuilt SDL library for
+us, instead of compiling it ourselves from the source, so – at least
+currently – there’s no way for the debugger to find associated sources.
+Let’s step-in again to get into the `SDL_RenderDrawColor`:
+
+![DevTools showing disassembly view of `mandelbrot.wasm`](/web/updates/images/2020/12/webassembly/image1.png)
+
+We’re back to the raw WebAssembly debugging experience.
+
+Now, it looks a bit scary and isn’t something most Web developers will
+ever need to deal with, but occasionally you might want to debug a
+library built without debug information – whether because it’s a
+3<sup>rd</sup>-party library you have no control over, or because you’re
+running into one of those bugs that occurs only on production.
+
+To aid in those cases, we’ve made some improvements to the basic
+debugging experience, too.
+
+First of all, if you used raw WebAssembly debugging before, you might
+notice that the entire disassembly is now shown in a single file – no
+more guessing which function a **Sources** entry “wasm-53834e3e/
+wasm-53834e3e-7” possibly corresponds to.
+
+### New name generation scheme {: #names }
+
+We improved names in the disassembly view, too. Previously you’d see
+just numeric indices, or, in case of functions, no name at all.
+
+Now we’re generating names similarly to other disassembly tools, by
+using hints from the [WebAssembly name
+section](https://webassembly.github.io/spec/core/appendix/custom.html#name-section),
+import/export paths and, finally, if everything else fails, generating
```

Comment:
```suggestion
`import`/`export` paths and, finally, if everything else fails, generating
```

Reply:
Those are not code quotes, just "import/export" as generic terms.

Reply:
OTOH... the corresponding code instructions are named that way too. I don't know, I'm happy to merge if you think this is better, but in the context it was meant as just a generic term.

Reply:
I think I'll leave this one as-is.